R/convergence.R

```diff
@@ -43,14 +43,25 @@ NULL
 #' mu <- extract_variable_matrix(example_draws(), "mu")
 #' rhat_basic(mu)
 #'
+#' d <- as_draws_rvars(example_draws("multi_normal"))
+#' rhat_basic(d$Sigma)
+#'
+#' @export
+rhat_basic <- function(x, split = TRUE) UseMethod("rhat_basic")
```

Comment:
Since we are now making a lot of these functions generic, I think we should talk about the argument structure of the generic. Most notably:

- Should the generics have a `...` argument to be more extensible by other packages?
- Should additional arguments such as `split` already be part of the generic, or should we rather go with the most generel `x, ...` signature?

Reply:
Good point. I think adding `...` makes sense for sure. I'm not sure which other arguments are necessary in the base generics, it seems like this might vary depending on the function?

Reply:
> Good point. I think adding ... makes sense for sure. I'm not sure which other arguments are necessary in the base generics, it seems like this might vary depending on the function?

I agree that adding `...` makes sense and that for other arguments it would depend on the function. For example, for `ess_quantile` 

```r
ess_quantile(x, probs = c(0.05, 0.95), names = TRUE)
```

I think it could make sense for `probs` to be in the generic because it's an essential argument but the `names` argument probably doesn't need to be part of the generic. That said, I'd also be fine with just going with the more general `x,...` signature and leaving all the other arguments for the methods.